### DISCUSSION TITLE
Blending on iOS

### DISCUSSION THREAD

**Discussion by rafisher79** [ExperienceLovelace/ha-floorplan#234] [Q&A]
My floor plan is working perfectly on the PC, but on iOS (both the app and in chrome) the color lighting function as well as the brightness doesn’t seem to be working. Before troubling everyone with the code, I wanted to see if this was a known limitation with iOS.

**Comment by OzGav** [comment]
I am not aware of any limitations or others having problems with IOS….

**Comment by rafisher79** [comment]
here's my code:

                service_data:
                  element: ${entity.entity_id.replace('light.', 'light_overlay.')}
                  style: |
                    >
                    if( entity.state !== "on" )
                        return "display: none;";
                    let hue = 0;
                    let sat = 0;
                    if( entity.attributes.hs_color )
                    {
                        hue = entity.attributes.hs_color[0];
                        sat = entity.attributes.hs_color[1];
                    }
                    if( sat < 10 )
                    {
                      return `
                        display: block;
                        filter:
                          opacity(calc( ${entity.attributes.brightness} / 255));`
                    }
                      return `
                        display: block;
                        filter:
                          sepia(100%)
                          hue-rotate(calc( ${hue}deg - 55deg ))
                          saturate(calc( ${sat}% * 2 ))
                          opacity(calc( ${entity.attributes.brightness} / 255));

I'm not sure why, but on the tablets/phones, the style isn't being returned - just on the desktop. I'm actually using this code on a traditional picture elements card with a PNG as opposed to the SVG and it works fine

**Comment by exetico** [comment]
Hi, 

One thing you can do is to enable debug in Chrome, and remote debug it. I've not faced any issues on my mobile.

Please provide one specific example of one specific function, and I'll give it a try. 

While doing so, please try on a Android-device if possible, and compare the version of Chrome on each of them.

Also, share the Google Chrome version here, for the device with problems.

**Reply by rafisher79** [reply]
No matter what I do, the light overlay won't change color based on hue-rotate

the code in questions is above, but it pertains specifically to an overlay I have in the SVG. On the PC, it works great and the overlay reads the element from the SVG with the correct style:

```
<svg id="svg8"....> (I removed the extra info)
<g id="background"> 
<g id="light-overlays">
<image id="light_overlay.hue_color_lamp_1" width="2300" height="1500" href="/local/assets/ha-floorplan/images/fp034.png" xlink:href="d:/HAtests/local/assets/ha-floorplan/images%5Cfp034.png" style="display: block; filter: sepia(100%) hue-rotate(214.663deg) saturate(139.608%); opacity: 0.368627;"></image>
> "
```

I think the iOS browsers and apps are having trouble specifically getting the filter variables from the element style and applying it to the image from the SVG, as it loses all the "filter" items, keeping the opacity though. Unfortnately, I can't get that level of inspection from my ipad to verify, but I think it may be a limitation of SVGs with iOS. (chrome 105.0.5195.129)

**Reply by exetico** [reply]
Why can't you "get that level of inspection" from your iPad?

**Reply by rafisher79** [reply]
I don’t know how to inspect individual elements on the iOS system. I’m fairly certain that it’s an Apple thing when it comes to SVG elements, as I was able to get it going on the Android tablet.

**Reply by exetico** [reply]
You mentioned it was a Google browser, right? And.. Did it fail on Android, too? Else I can't really see why it should be a Apple/iOS problem :smiley: . 

Isn't this possible anymore... or? 
https://medium.com/an-idea/google-chrome-remote-debug-with-ios-iphone-safari-to-windows-pc-d05ef653e98

That's how I'm doing it on Android :+1: ...

**Reply by rafisher79** [reply]
I’ll give that a shot this weekend to see what I can see. The layout did work properly on chrome on Android. On iOS it doesn’t work in chrome nor the app, so I still think it’s a goofy Apple thing. Either way, this is still so much better than picture elements because the coding is much cleaner for a house like mine with way too many entities. I can live with the iOS trouble. I’ll let you know what info I can gather with your directions on the remote debug.

**Reply by rafisher79** [reply]
OK, I inspected it on IOS and the tags are there, but it just doesn't want to actually use them. Like I said - its no big deal.

**Reply by RogerSelwyn** [reply]
> OK, I inspected it on IOS and the tags are there, but it just doesn't want to actually use them. Like I said - its no big deal.

Just wondering if you managed to fix this. Spent some time last night showing it works on an html img tag, but doesn't seem to work on a svg image tag.

**Reply by RogerSelwyn** [reply]
Came across this - https://codepen.io/starzonmyarmz/pen/KpMMGb - which allows you to try changing colour on a svg image. It works on a Mac, but doesn't work on iOS. By looking at it I could (dynamically) add a filter into the svg itself, but to be honest that is getting into the 'too hard' bucket.

**Comment by zacounts172** [comment]
Yea, I am having the same issue with iOS. Looks like the filter element attribute for the SVG is not being taken into account. 

I might as well just cut my losses and move to an Android Device like the Amazon Fire HD. Has anyone tried it on the Android Companion App? Does the blending work?

**Reply by exetico** [reply]
Are you able to reproduce the issue in Safari on MacOS?

**Reply by zacounts172** [reply]
Yes. Used the web inspector to check the HTML DOM code in Safari browser and when I tried to toggle the filter on and off, it had no effect. 

However, Chrome on MacOS worked fine .

**Reply by exetico** [reply]
Okay :) If so, please submit an issue just to keep track of it, and remember to link to the discussion.

**Reply by rafisher79** [reply]
Yes, It’s something with the way iOS reads the svg. I ended up buying a couple of fire tablets and using those as dashboards throughout the house and they work well.
